File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <stdio.h>
#include <iostream>
#include <string>
#include <stack>
#include <utility>
#include <algorithm>
int main(){
    std::cin.tie(0);
    std::ios::sync_with_stdio(0);
    int answer{}, n{};
    bool before=false;
    std::stack <bool> st{};
    std::string str{};
    std::cin >> str;
    for(int i=0; i<str.length();i++){
        if(str[i]=='('){
            n++;
            before=true;
        }
        else{
            n--;
            if(before){
                answer+=n;
            }
            else{
                answer++;
            }
            before=false;
        }
    }
    std::cout << answer;
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys
str=sys.stdin.readline().strip()
answer=0
n=0
before=False
for i in str:
    if i=='(':
        n+=1
        before=True
    else:
        n-=1
        if before:
            answer+=n
        else:
            answer+=1
        before=False
print(answer)
